shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -1484,6 +1484,83 @@ public static WireCommand readFrom(ByteBufInputStream in, int length) throws IOE
         }
     }
 
+    @Data
+    public static final class MergeSegmentsBatch implements Request, WireCommand {
+        final WireCommandType type = WireCommandType.MERGE_SEGMENTS_BATCH;
+        final long requestId;
+        final String targetSegmentId;
+        final List<String> sourceSegmentIds;
+        @ToString.Exclude
+        final String delegationToken;
+
+        @Override
+        public void process(RequestProcessor cp) {
+            cp.mergeSegmentsBatch(this);
+        }
+
+        @Override
+        public void writeFields(DataOutput out) throws IOException {
+            out.writeLong(requestId);
+            out.writeUTF(targetSegmentId);
+            out.writeInt(sourceSegmentIds.size());
+            for (int i = 0; i < sourceSegmentIds.size(); i++) {
+                out.writeUTF(sourceSegmentIds.get(i));
+            }
+            out.writeUTF(delegationToken == null ? "" : delegationToken);
+        }
+
+        public static WireCommand readFrom(DataInput in, int length) throws IOException {
+            long requestId = in.readLong();
+            String target = in.readUTF();
+            int sourceCount = in.readInt();
+            List<String> sources = new ArrayList<>(sourceCount);
+            for (int i = 0; i < sourceCount; i++) {
+                sources.add(in.readUTF());
+            }
+            String delegationToken = in.readUTF();
+            return new MergeSegmentsBatch(requestId, target, sources, delegationToken);
+        }
+    }
+
+    @Data
+    public static final class SegmentsMergedBatch implements Reply, WireCommand {
+        final WireCommandType type = WireCommandType.SEGMENTS_MERGED_BATCH;
+        final long requestId;
+        final String target;
+        final List<String> sources;
+        final List<Long> newTargetWriteOffset;
```

Comment:
Please add a comment that this may not be the offset immediately following the merge but is just SOME offset following the merge

Reply:
Okay